subsys/nfc/t4t/hl_procedure.c

```diff
@@ -0,0 +1,477 @@
+{
+	__ASSERT_NO_MSG(resp);
+
+	const u8_t *data = resp->data.buff;
+	u16_t len = resp->data.len;
+
+	if (len != NDEF_FILE_NLEN_SIZE) {
+		LOG_ERR("NDEF NLEN response is to long");
+		return -EINVAL;
+	}
+
+	t4t_hl.ndef.nlen = sys_get_be16(data);
+
+	return 0;
+}
+
+static int t4t_file_assign(u16_t id)
+{
+	struct nfc_t4t_tlv_block_file file;
+
+	__ASSERT_NO_MSG(t4t_hl.ndef.cc);
+
+	file.content = t4t_hl.ndef.buff;
+	file.len = t4t_hl.file_offset;
+
+	return nfc_t4t_cc_file_content_set(t4t_hl.ndef.cc, &file, id);
+}
+
+static int ndef_file_chunk_read(const struct nfc_t4t_apdu_resp *resp)
+{
+	__ASSERT_NO_MSG(resp);
+
+	int err;
+	u16_t file_id;
+	struct nfc_t4t_apdu_comm apdu_comm;
+	const u8_t *data = resp->data.buff;
+	u16_t len = resp->data.len;
+
+	if (t4t_hl.ndef.buff_size < t4t_hl.file_offset + len) {
+		return -ENOMEM;
+	}
+
+	memcpy(t4t_hl.ndef.buff + t4t_hl.file_offset, data, len);
+
+	t4t_hl.file_offset += len;
+
+	if (t4t_hl.file_offset < (t4t_hl.ndef.nlen + NDEF_FILE_NLEN_SIZE)) {
+		nfc_t4t_apdu_comm_clear(&apdu_comm);
+
+		apdu_comm.instruction = NFC_T4T_APDU_COMM_INS_READ;
+		apdu_comm.parameter = t4t_hl.file_offset;
+		apdu_comm.resp_len = MIN(t4t_hl.ndef.nlen - (t4t_hl.file_offset - NDEF_FILE_NLEN_SIZE),
+				MIN(APDU_LE_MAP_2_MAX_VALUE, t4t_hl.ndef.cc->max_rapdu_size));
+
+		t4t_hl.transaction_type = NFC_T4T_HL_NDEF_READ;
+
+		return t4t_hl_data_exchange(&apdu_comm);
+	}
+
+	file_id = sys_get_be16(t4t_hl.ndef.file_id);
+
+	err = t4t_file_assign(file_id);
+	if (err) {
+		return err;
+	}
+
+	if (hl_cb->ndef_read) {
+		hl_cb->ndef_read(file_id, t4t_hl.ndef.buff,
+				 t4t_hl.ndef.nlen + NDEF_FILE_NLEN_SIZE);
+	}
+
+	return 0;
+}
+
+static int ndef_file_chunk_update(void)
+{
+	struct nfc_t4t_apdu_comm apdu_comm;
+	u8_t nlen_data[NDEF_FILE_NLEN_SIZE];
+
+	nfc_t4t_apdu_comm_clear(&apdu_comm);
+
+	apdu_comm.instruction = NFC_T4T_APDU_COMM_INS_UPDATE;
+
+	if (t4t_hl.file_offset < t4t_hl.ndef.buff_size) {
+		apdu_comm.parameter = t4t_hl.file_offset;
+		apdu_comm.data.buff = t4t_hl.ndef.buff + t4t_hl.file_offset;
+		apdu_comm.data.len = MIN(t4t_hl.ndef.buff_size - t4t_hl.file_offset,
+				MIN(APDU_LE_MAP_2_MAX_VALUE, t4t_hl.ndef.cc->max_capdu_size));
+
+		t4t_hl.file_offset += apdu_comm.data.len;
+		t4t_hl.transaction_type = NFC_T4T_HL_NDEF_UPDATE;
+
+		return t4t_hl_data_exchange(&apdu_comm);
+	}
+
+	sys_put_be16(t4t_hl.ndef.nlen, nlen_data);
+
+	apdu_comm.parameter = 0;
+	apdu_comm.data.buff = nlen_data;
+	apdu_comm.data.len = sizeof(nlen_data);
+
+	t4t_hl.transaction_type = NFC_T4T_HL_NDEF_NLEN_UPDATE;
+
+	return t4t_hl_data_exchange(&apdu_comm);
+}
+
+static void on_ndef_nlen_update(void)
+{
+	u16_t file_id = sys_get_be16(t4t_hl.ndef.file_id);
+
+	if (hl_cb->ndef_updated) {
+		hl_cb->ndef_updated(file_id);
+	}
+}
+
+static int on_rapdu_succes(const struct nfc_t4t_apdu_resp *resp)
+{
+	int err = 0;
+
+	switch (t4t_hl.transaction_type) {
+	case NFC_T4T_HL_SELECT:
+		if (hl_cb->selected) {
+			hl_cb->selected(t4t_hl.select_type);
+		}
+
```

Comment:
these empty lines before break are not really needed and seems inconsistent with line 259

Reply:
Yes, no needed.